### PR TITLE
Fixups for the lightning_and_keras PR (#12)

### PR DESCRIPTION
**Pull request by plstcharles**
These changes introduce a bunch of fixups/refactorings to make the latest PyTorch-Lightning changes a bit cleaner. In short:
 - Moved the `load_loss` and `load_optimizer` functions to a new module (`optim`) so that they can be imported in the model itself;
 - Replaced the PyTorch dataloader getter by a proper PyTorch-Lightning Data Module getter (the latter is a better choice with PyTorch lightning in general as it allows more features in the trainer);
 - Removed the 'TraineeWrapper' object from the train module and replaced it by a proper LightningModule-based model that defines the required step functions --- it also holds the loss function and the optimizer, as preferred by PyTorch-Lightning design philosophy;
 - Removed the unnecessary model upload for PyTorch in the model getter (PyTorch-Lightning takes care of all that now).

I tested these changes by generating both PyTorch and Keras projects, testing the pre-commit hook, running pytest, and running the local example config.

### CODE REVIEW

File: {{cookiecutter.project_slug}}/{{cookiecutter.project_slug}}/models/my_model.py [hunk 1]
```diff
@@ -58,7 +76,36 @@ def forward(self, data):
...
+            batch: typing.Any,
+            batch_idx: int,
+    ) -> typing.Any:
+        """Runs the prediction + evaluation step for training/validation/testing."""
+        input_data, targets = batch
+        preds = self(input_data)  # calls the forward pass of the model
+        loss = self.loss_fn(preds, targets)
+        return loss
+
+    def training_step(self, batch, batch_idx):
```
Comment (mirkobronzi) [inline {{cookiecutter.project_slug}}/{{cookiecutter.project_slug}}/models/my_model.py:94]: I am not sure about this part (removing the trainee object and move the methods in the model itself.
The problem I see is that, if we have multiple models, the train/valid steps are most likely to be the same. This would lead to duplication of code.
I suppose a solution is to subclass models..
I am not sure which one makes more sense. Any previous experience on your side?

Comment (plstcharles) [reply {{cookiecutter.project_slug}}/{{cookiecutter.project_slug}}/models/my_model.py:94]: A very good point! For NRCAN, we're doing the subclassing. It seems ideal because we'd probably always work with models that all do the same task (e.g. "image segmentation!"), so the loop stuff that contains a bit of task-related logic can be in a base class that's shared across all of the models.

One way to make it more obvious for future users here could be to send the step implementations into a base class and add a derived class right away that contains the model building and optimization stuff. I can do that probably some time next week!

**Review by mirkobronzi — APPROVED** [review]
